tests/support/cluster_helper.tcl

```diff
@@ -0,0 +1,96 @@
+# Helper functions specifically for setting up and configuring redis
+# clusters.
+
+# Check if cluster configuration is consistent.
+proc cluster_config_consistent {} {
+    for {set j 0} {$j < [llength $::servers]} {incr j} {
+        if {$j == 0} {
+            set base_cfg [R $j cluster slots]
+        } else {
+            if {[R $j cluster slots] != $base_cfg} {
+                return 0
+            }
+        }
+    }
+
+    return 1
+}
+
+# Wait for cluster configuration to propagate and be consistent across nodes.
+proc wait_for_cluster_propagation {} {
+    wait_for_condition 50 100 {
+        [cluster_config_consistent] eq 1
+    } else {
+        fail "cluster config did not reach a consistent state"
+    }
+}
+
+# Check that cluster nodes agree about "state", or raise an error.
+proc wait_for_cluster_state {state} {
+    for {set j 0} {$j < [llength $::servers]} {incr j} {
+        wait_for_condition 100 50 {
+            [CI $j cluster_state] eq $state
+        } else {
+            fail "Cluster node $j cluster_state:[CI $j cluster_state]"
+        }
+    }
+}
+
+# Default slot allocation for clusters, each primary has a continuous block
+# and approximately equal number of slots.
+proc continuous_slot_allocation {primaries} {
+    set avg [expr double(16384) / $primaries]
+    set slot_start 0
+    for {set j 0} {$j < $primaries} {incr j} {
+        set slot_end [expr int(ceil(($j + 1) * $avg) - 1)]
+        R $j cluster addslotsrange $slot_start $slot_end
+        set slot_start [expr $slot_end + 1]
+    }
+}
+
+# Setup method to be executed to configure the cluster before the
+# tests run.
+proc cluster_setup {primaries node_count slot_allocator code} {
+    # Have all nodes meet
+    for {set i 1} {$i < $node_count} {incr i} {
+        R 0 CLUSTER MEET [srv -$i host] [srv -$i port]
+    }
+
+    $slot_allocator $primaries
+
+    wait_for_cluster_propagation
+
+    # Setup primary/replica relationships
+    for {set i 0} {$i < $primaries} {incr i} {
```

Comment:
the terminology in redis (for now) is "master".
i think that starting to add use of "primary" in random places is confusing.

Reply:
I'll pushback that primary is used a couple of places, and honestly I don't want to use the term master when it's not strictly required. I'll start the wider conversation we as a group need to have about replacing the term if I need to.

Reply:
my point is that until we decide to replace the term (and come up with a plan of how we do that), we shouldn't start introducing confusing terminology change to the code.
i noticed it already slipped in to a few places in the code base, but i see that an unintentional and local change.

p.s. we didn't even finish the transition of the slave term yet (only changed inputs but not outputs, e.g. INFO)

Reply:
Hmm, ok, I'll start the conversation then, since I want it to happen.